Kickstarter-iOS/DataSources/CommentRepliesDataSource.swift

```diff
@@ -5,25 +5,63 @@ import UIKit
 
 internal final class CommentRepliesDataSource: ValueCellDataSource {
   internal enum Section: Int {
-    case comment
-    case replies
+    case rootComment
+    case comments
+    case empty
+    case error
   }
 
   internal func createContext(comment: Comment) {
-    let section = Section.comment.rawValue
+    let section = Section.rootComment.rawValue
     self.clearValues()
 
     self.appendRow(value: comment, cellClass: RootCommentCell.self, toSection: section)
   }
 
-  internal func load(replies: [Comment], project: Project) {
-    replies.forEach { reply in
-      self.appendRow(
-        value: (reply, project),
-        cellClass: CommentCell.self,
-        toSection: Section.replies.rawValue
-      )
+  /// Loads in a new page of comments.
+  internal func load(comments: [Comment], project: Project) {
+    // Clear all but rootComment section.
+    self.clearValues(section: Section.empty.rawValue)
+    self.clearValues(section: Section.error.rawValue)
+
+    self.padValuesForSection(Section.comments.rawValue)
+
+    let current = (self[section: Section.comments.rawValue] as? [(comment: Comment, project: Project)]) ?? []
+
+    self.clearValues(section: Section.comments.rawValue)
+
+    let newComments = comments.map { comment in
+      (comment, project)
     }
+
+    let allComments = newComments + current
+
+    allComments.forEach { comment, project in
+      self.loadValue(comment, project: project)
+    }
+  }
+
+  // TODO: Implement and write tests
+  internal func showEmptyState() {
```

Comment:
Deleting this as there is no empty state for replies. If there are no replies, the `View Replies` button is hidden.